src/assets/wise5/components/peerChat/peer-chat-question-bank/peer-chat-question-bank.component.ts

```diff
@@ -44,38 +47,61 @@ export class PeerChatQuestionBankComponent implements OnInit {
-      const cRaterResponses = peerGroupStudentData.map((peerMemberData: PeerGroupStudentData) => {
-        return new CRaterResponse({
-          ideas: peerMemberData.annotation.data.ideas,
-          scores: peerMemberData.annotation.data.scores,
-          submitCounter: peerMemberData.studentWork.studentData.submitCounter
-        });
-      });
-      const feedbackRuleEvaluator = new FeedbackRuleEvaluator(
-        new FeedbackRuleComponent(
-          this.content.questionBank.getRules(),
-          (referenceComponent.content as OpenResponseContent).maxSubmitCount,
-          false
-        )
+      const questionBankRules = this.chooseQuestionBankRulesToDisplay(
+        referenceComponent,
+        peerGroupStudentData
       );
-      const feedbackRule: QuestionBankRule = feedbackRuleEvaluator.getFeedbackRule(
-        cRaterResponses
-      ) as QuestionBankRule;
-      this.questions = feedbackRule.questions;
-      this.displayedQuestionBankRuleChange.emit(feedbackRule);
+      this.displayedQuestionBankRules = questionBankRules;
+      this.displayedQuestionBankRulesChange.emit(questionBankRules);
+      this.setQuestions();
+    });
+  }
+
+  private chooseQuestionBankRulesToDisplay(
+    referenceComponent: WISEComponent,
+    peerGroupStudentData: PeerGroupStudentData[]
+  ): QuestionBankRule[] {
+    const cRaterResponses = peerGroupStudentData.map((peerMemberData: PeerGroupStudentData) => {
+      return new CRaterResponse({
+        ideas: peerMemberData.annotation.data.ideas,
+        scores: peerMemberData.annotation.data.scores,
+        submitCounter: peerMemberData.studentWork.studentData.submitCounter
+      });
     });
+    const feedbackRuleEvaluator = new FeedbackRuleEvaluator(
+      new FeedbackRuleComponent(
+        this.content.questionBank.getRules(),
+        (referenceComponent.content as OpenResponseContent).maxSubmitCount,
+        false
+      )
+    );
+    return this.filterQuestions(
+      feedbackRuleEvaluator.getFeedbackRules(cRaterResponses) as QuestionBankRule[]
+    );
+  }
+
+  private filterQuestions(questionBankRules: QuestionBankRule[]): QuestionBankRule[] {
+    const filteredRules: QuestionBankRule[] = JSON.parse(JSON.stringify(questionBankRules));
+    filteredRules.forEach((rule) => (rule.questions = []));
+    let numAdded = 0;
+    let ruleIndex = 0;
+    const totalNumQuestions = questionBankRules.map((rule) => rule.questions).flat().length;
+    const maxQuestionsToShow = this.content.questionBank.maxQuestionsToShow;
+    while (numAdded < maxQuestionsToShow && numAdded != totalNumQuestions) {
+      if (questionBankRules[ruleIndex].questions.length > 0) {
+        const question = questionBankRules[ruleIndex].questions.shift();
+        filteredRules[ruleIndex].questions.push(question);
+        numAdded++;
+      }
+      ruleIndex = (ruleIndex + 1) % questionBankRules.length;
+    }
+    return filteredRules.filter((rule) => rule.questions.length > 0);
```

Comment:
Would it be better to pass in maxQuestionsToShow as a parameter to filterQuestions() so it can be a pure function?

```JSON.parse(JSON.stringify(questionBankRules))```
can be changed to
```this.utilService.makeCopyOfJSONObject(questionBankRules)```

Reply:
I don't think we should add a UtilService dependency to this class just to make a copy of the object.

Reply:
Calling JSON.parse(JSON.stringify()) is adding duplication. The point of the UtilService is to provide common functions that our components use.